src/modules/command/statement.rs

```diff
@@ -49,6 +49,9 @@ impl TranslateModule for CommandStatement {
         let interps = self.interps.iter()
             .map(|item| item.translate(meta))
             .collect::<Vec<String>>();
+
+        scan_append_externs(self.strings.clone(), meta);
```

Comment:
I think this is probably not something we want to try and do. Trying to understand all of the dependencies that a script relies on is a big ask. There could be dependencies that are set based on environment variables, the script could change the `$PATH` adding new things that were previously not present.

I think that in this initial pass we should add support for the Amber runtime dependencies. If we really want to provide this affordance to Amber users then it seems like it would be best to add a library function they can call to explicitly call out a dependency they want to check for.

In that way the script checks for the pieces that it depends on and then it provides a way for script uathors to do something siimlar.

Reply:
Oh yeah. That's actually a good point that people could add things later on in the script to the `$PATH`. For now let's just leave it as a precheck for making sure that Amber script can at least run.